src/resources/js/aggregator-fields.js

```diff
@@ -0,0 +1,358 @@
+var tribe_ea = tribe_ea || {};
+
+// Setup the global Variable
+tribe_ea.fields = {
+	// Store the Required Selectors
+	slct: {
+		container: '.tribe-ea',
+		help: '.tribe-ea-help',
+		dependent: '.tribe-ea-dependent',
+		active: '.tribe-ea-active',
+		fields: '.tribe-ea-field',
+		dropdown: '.tribe-ea-dropdown',
+		media_button: '.tribe-ea-media_button',
+		datepicker: '.tribe-ea-datepicker'
+	},
+
+	media: {},
+
+	// Store the jQuery elements
+	$: {},
+
+	// Store the methods for creating the fields
+	construct: {},
+
+	// Store L10N strings
+	l10n: window.tribe_l10n_ea_fields
+};
+
+( function( $, _, my ) {
+	'use strict';
+
+	/**
+	 * Sets up the fields for EA pages
+	 *
+	 * @return void
+	 */
+	my.init = function() {
+		my.$.container = $( my.slct.container );
+
+		// Fetch All the Help Icons
+		my.$.help = my.$.container.find( my.slct.help );
+		my.$.help.bumpdown();
+
+		// Update what fields we currently have to setup
+		my.$.fields = my.$.container.find( my.slct.fields );
+
+		// Setup each type of field
+		$.each( my.construct, function( key, callback ){
+			callback( my.$.fields );
+		} );
+
+		my.$.fields.on( 'change', function( e ) {
```

Comment:
I will move this structure to another place, more specifically another jQuery function file. To allow more flexibility, and usage around the plugin.